frontend/src/pages/Sessions/SessionsFeedV3/SessionQueryBuilder/components/SessionFeedConfigurationV2/SessionFeedConfigurationV2.tsx

```diff
@@ -40,51 +41,84 @@ export const DropdownMenu = function ({
 		setShowDetailedSessionView,
 	} = usePlayerConfiguration()
 	const sessionFeedConfiguration = useSessionFeedConfiguration()
+	const [open, setOpen] = useState(false)
 
-	const [open, setOpen] = React.useState(false)
+	const menuRef = useRef<HTMLDivElement>(null)
+	const buttonRef = useRef<HTMLButtonElement>(null)
+
+	const [delayClose, setDelayClose] = useState(false)
+
+	useWindowEvent('click', (event) => {
+		if (
+			!(
+				isInsideElement(event, menuRef.current) ||
+				isInsideElement(event, buttonRef.current)
+			) &&
+			!delayClose
```

Comment:
A bit hacky, but using ref for submenus did not work (because the element was destroyed by the time the listener fires)